packages/liveblocks-core/src/http-client.ts

```diff
@@ -1,11 +1,1400 @@
+    type: TextEditorType;
+    rootKey: string;
+  }): Promise<void>;
+
+  listTextVersions({ roomId }: { roomId: string }): Promise<{
+    versions: {
+      type: "historyVersion";
+      kind: "yjs";
+      id: string;
+      authors: {
+        id: string;
+      }[];
+      createdAt: Date;
+    }[];
+    requestedAt: Date;
+  }>;
+
+  listTextVersionsSince({
+    roomId,
+    since,
+    signal,
+  }: {
+    roomId: string;
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    versions: {
+      type: "historyVersion";
+      kind: "yjs";
+      id: string;
+      authors: {
+        id: string;
+      }[];
+      createdAt: Date;
+    }[];
+    requestedAt: Date;
+  }>;
+
+  streamStorage(options: {
+    roomId: string;
+    getAuthValue: () => Promise<AuthValue>;
+  }): Promise<IdTuple<SerializedCrdt>[]>;
+
+  sendMessages<P extends JsonObject, E extends Json>(options: {
+    roomId: string;
+    getAuthValue: () => Promise<AuthValue>;
+    nonce: string | undefined;
+    messages: ClientMsg<P, E>[];
+  }): Promise<Response>;
+}
+
+export interface NotificationHttpApi {
+  getInboxNotifications<M extends BaseMetadata>(options?: {
+    cursor?: string;
+  }): Promise<{
+    inboxNotifications: InboxNotificationData[];
+    threads: ThreadData<M>[];
+    nextCursor: string | null;
+    requestedAt: Date;
+  }>;
+
+  getInboxNotificationsSince<M extends BaseMetadata>(options: {
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    inboxNotifications: {
+      updated: InboxNotificationData[];
+      deleted: InboxNotificationDeleteInfo[];
+    };
+    threads: {
+      updated: ThreadData<M>[];
+      deleted: ThreadDeleteInfo[];
+    };
+    requestedAt: Date;
+  }>;
+
+  getUnreadInboxNotificationsCount(): Promise<number>;
+
+  markAllInboxNotificationsAsRead(): Promise<void>;
+
+  markInboxNotificationAsRead(inboxNotificationId: string): Promise<void>;
+
+  deleteAllInboxNotifications(): Promise<void>;
+
+  deleteInboxNotification(inboxNotificationId: string): Promise<void>;
+}
+
+export interface ClientHttpApi extends RoomHttpApi, NotificationHttpApi {
+  getUserThreads_experimental<M extends BaseMetadata>(options?: {
+    cursor?: string;
+    query?: {
+      resolved?: boolean;
+      metadata?: Partial<QueryMetadata<M>>;
+    };
+  }): Promise<{
+    threads: ThreadData<M>[];
+    inboxNotifications: InboxNotificationData[];
+    nextCursor: string | null;
+    requestedAt: Date;
+    permissionHints: Record<string, Permission[]>;
+  }>;
+
+  getUserThreadsSince_experimental<M extends BaseMetadata>(options: {
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    inboxNotifications: {
+      updated: InboxNotificationData[];
+      deleted: InboxNotificationDeleteInfo[];
+    };
+    threads: {
+      updated: ThreadData<M>[];
+      deleted: ThreadDeleteInfo[];
+    };
+    requestedAt: Date;
+    permissionHints: Record<string, Permission[]>;
+  }>;
+}
+
+export function createHttpClient({
```

Comment:
We need to think about naming here a bit. `new HttpClient` and `createHttpClient` sound similar, but one creates an abstract thing that can be used to query (any) backend endpoint, and one uses that thing to expose type-safe helpers.

Maybe rename `createHttpClient` to `createRoomAPI` or `createRoomBackendAPI` or something?

Reply:
+1, I also find the name "HTTP client" too confusing for what it does, in my ears it's something super low-level like a `fetch` alternative API.